### PR TITLE
📝 Add documentation for `model_config`

### PR DESCRIPTION
**Pull request by Kludex**
For review, check the preview page, please. 👍

https://ba40c80e.pydantic-docs2.pages.dev/usage/model_config/ 

Selected Reviewer: @samuelcolvin

### CODE REVIEW

**Comment by Kludex** [comment]
please review

File: docs/usage/model_config.md [hunk 1]
```diff
@@ -276,91 +331,142 @@ print(Character.model_json_schema(by_alias=True))
...
+
+my_sub_user = SubUser(hobbies=['scuba diving'], sins=['lying'])
+t = Transaction(user=my_sub_user)
+print(t)
+#> user=SubUser(hobbies=['scuba diving'], sins=['lying'])
 ```
 
-To prevent this, you can enable `Config.smart_union`. _Pydantic_ will then check all allowed types before even trying to coerce.
-Know that this is of course slower, especially if your `Union` is quite big.
+1. `revalidate_instances` is set to `'never'` by **default.
```
Comment (tpdorsey) [inline docs/usage/model_config.md:372]:
```suggestion
1. `revalidate_instances` is set to `'never'` by default.
```

File: docs/usage/model_config.md [hunk 2]
```diff
@@ -67,6 +71,26 @@ except ValidationError as e:
     """
 ```
 
+
+1. If using the `dataclass` from the standard library, you should use `__pydantic_config__` instead.
```
[the remark this answers is not in the document]
Comment (Kludex) [reply docs/usage/model_config.md:75]: That's why I suggested to check the preview page on the description. 😅

[the remark this answers is not in the document]
Comment (Kludex) [reply docs/usage/model_config.md:75]: https://ba40c80e.pydantic-docs2.pages.dev/usage/model_config/

**Pull request merged**